test/common/common.go

```diff
@@ -0,0 +1,446 @@
+	return testService(name, namespace, corev1.ServiceTypeLoadBalancer)
+}
+
+func testService(name, namespace string, svcType corev1.ServiceType) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"external-dns.mydomain.org/publish": "yes",
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"name": "hello-openshift",
+			},
+			Type: svcType,
+			Ports: []corev1.ServicePort{
+				{
+					Protocol: corev1.ProtocolTCP,
+					Port:     80,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 8080,
+					},
+				},
+			},
+		},
+	}
+}
+
+func MustGetEnv(name string) string {
+	val := os.Getenv(name)
+	if val == "" {
+		panic(fmt.Sprintf("environment variable %s must be set", name))
+	}
+	return val
+}
+
+func RootCredentials(kubeClient client.Client, name string) (map[string][]byte, error) {
+	secret := &corev1.Secret{}
+	secretName := types.NamespacedName{
+		Name:      name,
+		Namespace: "kube-system",
+	}
+	if err := kubeClient.Get(context.TODO(), secretName, secret); err != nil {
+		return nil, fmt.Errorf("failed to get credentials secret %s: %w", secretName.Name, err)
+	}
+	return secret.Data, nil
+}
+
+func LookupARecord(host, server string) ([]string, error) {
+	dnsClient := &dns.Client{}
+	message := &dns.Msg{}
+	message.SetQuestion(dns.Fqdn(host), dns.TypeA)
+	response, _, err := dnsClient.Exchange(message, fmt.Sprintf("%s:53", server))
+	if err != nil {
+		return nil, err
+	}
+	if len(response.Answer) == 0 {
+		return nil, fmt.Errorf("not found")
+	}
+	var ips []string
+	for _, ans := range response.Answer {
+		if aRec, ok := ans.(*dns.A); ok {
+			ips = append(ips, aRec.A.String())
+		}
+	}
+	if len(ips) == 0 {
+		return nil, fmt.Errorf("not found")
+	}
+	return ips, nil
+}
+
+// DnsQueryClusterInternal queries for a DNS hostname inside the VPC of a cluster using a dig pod. It returns a
+// map structure of the resolved IPs for easy lookup.
+func DnsQueryClusterInternal(ctx context.Context, t *testing.T, cl client.Client, cs *kubernetes.Clientset, namespace, hostname string) (map[string]struct{}, error) {
+	t.Helper()
+
+	clientPod := buildDigPod("digpod", namespace, hostname)
+	if err := cl.Create(ctx, clientPod); err != nil {
+		return nil, fmt.Errorf("failed to create pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+	}
+	defer func() {
+		_ = cl.Delete(ctx, clientPod)
+	}()
+
+	// Loop until dig pod starts, then parse logs for query results.
+	var responseCode string
+	var gotIPs map[string]struct{}
+	err := wait.PollUntilContextTimeout(ctx, 5*time.Second, 10*time.Minute, true, func(ctx context.Context) (bool, error) {
+		if err := cl.Get(ctx, types.NamespacedName{Name: clientPod.Name, Namespace: clientPod.Namespace}, clientPod); err != nil {
+			t.Errorf("Failed to get pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
```

Comment:
Same polling question as before. Is this error fatal or we would like to retry? We have different answers for different pollings.

Reply:
Yea sorry I think I misunderstood `t.Errorf`. Fixed.